src/lib/mongoBackend/mongoQueryTypes.cpp

```diff
@@ -264,7 +288,20 @@ HttpStatusCode mongoEntityTypes
       }
     }
 
-    responseP->entityTypeVector.push_back(entityType);
+    // entityType corresponding to nullId case is skipped, as it is (eventually) added outside the for loop
+    if (!nullId)
+    {      
+      responseP->entityTypeVector.push_back(entityType);
+    }
+  }
+
+  if (emptyEntityTypeUsed)
+  {
+    responseP->entityTypeVector.push_back(emptyEntityType);
+  }
```

Comment:
Both results for entities without type (`null` in the results array) and type "" ("" in the results array) are aggregated into a single entry "" in the response to get types operation. Thus, I cannot push_back inside the for loop (as with a regular type) because that could lead to duplicated entries for "" (probably that would generate ilegal JSON). Thus, pushing back y deferred after the for loop.